src/mlpack/core/util/arma_traits.hpp

```diff
@@ -37,6 +37,18 @@ struct IsVector
   const static bool value = false;
 };
 
+template<typename MatType>
+struct IsMat
```

Comment:
```suggestion
struct IsMatrix
```

This is trivial, but I'd suggest `IsMatrix` instead of `IsMat` because it'll better match the (full) names of `IsVector` and `IsCube`.